core/modules/search/search.module

```diff
@@ -111,13 +111,94 @@ function search_block_info() {
   return $blocks;
 }
 
+/**
+ * Implements hook_block_configure().
+ */
+function search_block_configure($delta = '', $settings = array()) {
+  $form = array();
+  if ($delta == 'form') {
+    $search_settings = $settings['search'];
+    $search_settings += array(
+      'label' => '',
+      'label_hide' => TRUE,
+      'placeholder' => '',
+      'button' => 'Search',
+    );
+
+    $form['search'] = array(
+      '#type' => 'fieldset',
+      '#title' => t('Search form settings'),
+      '#collapsible' => TRUE,
+    );
+
+    $form['search']['label'] = array(
+      '#type' => 'textfield',
+      '#title' => t('Search box label'),
+      '#default_value' => check_plain($search_settings['label']),
+    );
+    $form['search']['label_hide'] = array(
+      '#type' => 'checkbox',
+      '#title' => t('Hide the search box label'),
+      '#description' => t('The search box label will always be visible to screen readers.'),
+      '#default_value' => $search_settings['label_display'],
+    );
+    $form['search']['placeholder'] = array(
+      '#type' => 'textfield',
+      '#title' => t('Placeholder text'),
+      '#default_value' => check_plain($search_settings['placeholder']),
+      '#description' => t('This text will appear in the search box until text is entered.'),
+    );
+    $form['search']['button'] = array(
+      '#type' => 'textfield',
+      '#title' => t('Button text'),
+      '#default_value' => check_plain($search_settings['button']),
+      '#description' => t('Leave this field empty to hide the submit button.'),
+    );
+  }
+  return $form;
+}
+
 /**
  * Implements hook_block_view().
  */
-function search_block_view($delta = '') {
+function search_block_view($delta = '', $settings = array(), $contexts = array()) {
   if (user_access('search content')) {
+    $search_settings = $settings['search'];
+    $settings += array(
+      'label' => '',
+      'label_hide' => TRUE,
+      'placeholder' => '',
+      'button' => 'Search',
+    );
+    $form = backdrop_get_form('search_block_form');
+
+    // Show the label.
+    if (!$search_settings['label_hide']) {
```

Comment:
if the "search box label will always be visible to screen readers", then shouldnt there always be one? are we missing an `else`?

Reply:
Hm. Yes, good point!

Reply:
In the function [`search_box()`](https://api.backdropcms.org/api/backdrop/core%21modules%21search%21search.module/function/search_box/1) the title display is already defined as `invisible`. Only when the title is visible does value needs to be changed to